### PR TITLE
MLDB-1793 try builtin fct

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: sql/sql_expression_operations.cc [hunk 1]
```diff
@@ -2256,21 +2256,48 @@ bindBuiltinFunction(SqlBindingScope & scope,
...
+                std::vector<ExpressionValue> evaluatedArgs;
+                evaluatedArgs.reserve(boundArgs.size());
+                try {
+                    evaluatedArgs.emplace_back(std::move(
+                        boundArgs[0](row, fn.filter)));
+                }
+                catch (const ML::Exception & exc) {
+                    evaluatedArgs.emplace_back(std::move(
+                        ExpressionValue::exception(exc.what(),
+                                                    Date::now())));
```
Comment (FinchPowers) [inline sql/sql_expression_operations.cc]: Le message de l'exception est conservé ici, mais ignoré ensuite par le builtin try. On peut le garder si on veut.

File: sql/sql_expression_operations.cc [hunk 2]
```diff
@@ -2256,21 +2256,48 @@ bindBuiltinFunction(SqlBindingScope & scope,
...
-                        evaluatedArgs.emplace_back(std::move(a(row, fn.filter)));
+            {
+                ExcAssert(args.size() == 2);
+                std::vector<ExpressionValue> evaluatedArgs;
+                evaluatedArgs.reserve(boundArgs.size());
+                try {
+                    evaluatedArgs.emplace_back(std::move(
+                        boundArgs[0](row, fn.filter)));
+                }
+                catch (const ML::Exception & exc) {
```
Comment (jeremybarnes) [inline sql/sql_expression_operations.cc]: The HttpReturnException has a lot more information within it, which we should extract here.  Or at least use ML::getExceptionString().

File: sql/sql_expression_operations.cc [hunk 3]
```diff
@@ -2256,21 +2256,48 @@ bindBuiltinFunction(SqlBindingScope & scope,
                 this,
                 fn.resultInfo};
     }
-    else {
+
+    if (functionName == "try") {
```
Comment (jeremybarnes) [inline sql/sql_expression_operations.cc]: This is the wrong place to put it.  We shouldn't have logic hardcoded into the generic function wrapper here.

**Comment by jeremybarnes** [comment]
I'm not convinced that a new atom type for exceptions is right.  We will then need to define the semantics of operations on them.  Why not make it much simpler, with the try function as a pure builtin that trys and catches within its lambda?

File: sql/sql_expression_operations.cc (hunk 3 above)
Comment (FinchPowers) [inline sql/sql_expression_operations.cc]: Étant donné que la call stack ne passe pas par les fonctions tant que les arguments ne sont pas prêts, je c'était le plus simple à faire. Ça pourrait être l'exception qui confirme la règle...

Autrement, as-tu une approche à me suggérer?

File: sql/sql_expression_operations.cc (hunk 2 above)
Comment (FinchPowers) [inline sql/sql_expression_operations.cc]: If the thrown exception was not of type HttpReturnException, will it still be caught?

**Comment by FinchPowers** [comment]
Replaced by https://github.com/mldbai/mldb/pull/525

**Pull request closed without merging**